src/drivers/indexeddb.js

```diff
@@ -1,766 +1,722 @@
-    //
-    // Furthermore, they have some outstanding bugs where blobs occasionally
-    // are read by FileReader as null, or by ajax as 404s.
-    //
-    // Sadly we use the 404 bug to detect the FileReader bug, so if they
-    // get fixed independently and released in different versions of Chrome,
-    // then the bug could come back. So it's worthwhile to watch these issues:
-    // 404 bug: https://code.google.com/p/chromium/issues/detail?id=447916
-    // FileReader bug: https://code.google.com/p/chromium/issues/detail?id=447836
-    //
-    function _checkBlobSupportWithoutCaching(idb) {
-        return new Promise(function(resolve, reject) {
-            var blob = _createBlob([''], {type: 'image/png'});
-            var txn = idb.transaction([DETECT_BLOB_SUPPORT_STORE], 'readwrite');
-            txn.objectStore(DETECT_BLOB_SUPPORT_STORE).put(blob, 'key');
-            txn.oncomplete = function() {
-                // have to do it in a separate transaction, else the correct
-                // content type is always returned
-                var blobTxn = idb.transaction([DETECT_BLOB_SUPPORT_STORE],
-                    'readwrite');
-                var getBlobReq = blobTxn.objectStore(
-                    DETECT_BLOB_SUPPORT_STORE).get('key');
-                getBlobReq.onerror = reject;
-                getBlobReq.onsuccess = function(e) {
-
-                    var storedBlob = e.target.result;
-                    var url = URL.createObjectURL(storedBlob);
-
-                    _blobAjax(url).then(function(res) {
-                        resolve(!!(res && res.type === 'image/png'));
-                    }, function() {
-                        resolve(false);
-                    }).then(function() {
-                        URL.revokeObjectURL(url);
-                    });
-                };
+            reject({status: xhr.status, response: xhr.response});
+        };
+        xhr.send();
+    });
+}
+
+//
+// Detect blob support. Chrome didn't support it until version 38.
+// In version 37 they had a broken version where PNGs (and possibly
+// other binary types) aren't stored correctly, because when you fetch
+// them, the content type is always null.
+//
+// Furthermore, they have some outstanding bugs where blobs occasionally
+// are read by FileReader as null, or by ajax as 404s.
+//
+// Sadly we use the 404 bug to detect the FileReader bug, so if they
+// get fixed independently and released in different versions of Chrome,
+// then the bug could come back. So it's worthwhile to watch these issues:
+// 404 bug: https://code.google.com/p/chromium/issues/detail?id=447916
+// FileReader bug: https://code.google.com/p/chromium/issues/detail?id=447836
+//
+function _checkBlobSupportWithoutCaching(idb) {
+    return new Promise(function(resolve, reject) {
+        var blob = createBlob([''], {type: 'image/png'});
+        var txn = idb.transaction([DETECT_BLOB_SUPPORT_STORE], 'readwrite');
```

Comment:
This stands for `transaction`, yeah? I'll go through and expand this, but I am afraid of variable names without vowels 😉